src/http/http.ml

```diff
@@ -438,36 +452,36 @@ let serve_with_details
      be pattern matching on the exception (but that might introduce dependency
      coupling), or the upstream should be patched to distinguish the errors in
      some useful way. *)
-  let httpaf_connection_handler client_address socket =
-    Lwt.catch
-      (fun () ->
-        httpaf_connection_handler client_address socket)
-      (fun exn ->
-        tls_error_handler client_address exn;
-        Lwt.return_unit)
+  let httpaf_connection_handler flow client_address =
+    try
+      httpaf_connection_handler client_address flow
+    with exn ->
+      tls_error_handler client_address exn
   in
 
-  (* Look up the low-level address corresponding to the interface. Hopefully,
-     this is a local interface. *)
-  let%lwt addresses = Lwt_unix.getaddrinfo interface (string_of_int port) [] in
-  match addresses with
-  | [] ->
-    Printf.ksprintf failwith "Dream.%s: no interface with address %s"
-      caller_function_for_error_messages interface
-  | address::_ ->
-  let listen_address = Lwt_unix.(address.ai_addr) in
-
-
-  (* Bring up the HTTP server. Wait for the server to actually get started.
-     Then, wait for the ~stop promise. If the ~stop promise ever resolves, stop
-     the server. *)
-  let%lwt server =
-    Lwt_io.establish_server_with_client_socket
-      listen_address
-      httpaf_connection_handler in
+  let listen_address =
+    (* Look up the low-level address corresponding to the interface. Hopefully,
+       this is a local interface. *)
+    let addresses = Unix.getaddrinfo interface (string_of_int port) [] in
```

Comment:
Note: Eio has [getaddrinfo_stream](https://ocaml-multicore.github.io/eio/eio/Eio/Net/index.html#val-getaddrinfo_stream) now.

Reply:
Just for my information, module `Unix` still does old-fashioned blocking I/O like it always did, right? This call was here because Eio temporarily didn't have a replacement for `getaddrinfo`.

Reply:
Correct.